src/services/task-processor.spec.js

```diff
@@ -156,6 +160,55 @@ describe('task-processor', () => {
       });
     });
 
+    describe('when task processing fails with an irrecoverable error', () => {
+      let expectedError;
+
+      beforeEach(async () => {
+        ctx = { cancellationRequested: false };
+
+        nextTask = { _id: taskId, taskType: TASK_TYPE.documentValidation, processed: false, attempts: [] };
+
+        lockStore.takeTaskLock.resolves(lock);
+        taskStore.getUnprocessedTaskById.resolves(nextTask);
+        expectedError = new Error('Processing failure 1');
+        expectedError.isIrrecoverable = true;
+        documentValidationTaskProcessor.process.rejects(expectedError);
+
+        await sut.process(taskId, batchParams, ctx);
+      });
+
+      it('should call lockStore.takeTaskLock', () => {
+        sinon.assert.calledOnce(lockStore.takeTaskLock);
+      });
+
+      it('should call taskStore.getUnprocessedTaskById', () => {
+        sinon.assert.calledWith(taskStore.getUnprocessedTaskById, taskId);
+      });
+
+      it('should call documentValidationTaskProcessor.process', () => {
+        sinon.assert.calledWith(documentValidationTaskProcessor.process, nextTask, batchParams, ctx);
```

Comment:
Maybe throw in here also a `calledOnce` assert, even though it is implied. by the lock being called once, I feel this would be clearer?

Reply:
Done!